tests/test_taar_similarity.py

```diff
@@ -147,40 +137,52 @@ def multi_clusters_df(generate_data):
             # Generate some valid addons for each client. Each stub
             # addon as a specific GUID that contains both the variation
             # id and the addon id for simplified debugging.
-            variation["active_addons"] = [{}]
+            variation["active_addons"] = []
             for addon_id in range(0, 3):
                 addon_name = "var-{}-guid-{}".format(variation_id, addon_id)
-                variation["active_addons"][0][addon_name] = {
```

Comment:
cc @mlopatka 

I don't think this test case wasn't written correctly before.  The `active_addons` field is defined as : 

```
    StructField("active_addons",         ArrayType(
        MapType(StringType(), StructType([
            StructField("blocklisted",      BooleanType(), True),
            StructField("type",             StringType(), True),
            StructField("signed_state",     LongType(), True),
            StructField("user_disabled",    BooleanType(), True),
            StructField("app_disabled",     BooleanType(), True),
            StructField("is_system",        BooleanType(), True),
            StructField("foreign_install",  BooleanType(), True)
        ]), True), True)),
```

which is roughly "list of dictionaries containing a dict that is `{addon_guid: {addon_metadata_dict}}`.

The loop on line 151 (was 141) now appends the dictionaries to the active_addons list.

Reply:
Bah.  Ignore https://github.com/mozilla/python_mozetl/pull/265#discussion_r227487130

On testing with live data on analysis.telemetry.mozilla.org, the active_addons column is a list of dictionaries of metadata.  There is no intermediary MapType involved.